index.js

```diff
@@ -2,30 +2,69 @@ import execa from 'execa';
 import pMemoize from 'p-memoize';
 import delay from 'delay';
 
+const powershell = (async (command, args) => {
+	if (typeof (command) !== 'string') {
+		throw new TypeError('Invalid command');
+	}
+
+	return execa(command, args, {shell: 'powershell'});
+});
+
 const getDevice = pMemoize(async () => {
-	if (process.platform !== 'darwin') {
-		throw new Error('macOS only');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+		const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
+
+		return result.groups.device;
 	}
 
-	const {stdout} = await execa('networksetup', ['-listallhardwareports']);
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapter', ['-Name Wi-Fi | Format-List -Property PnPDeviceID']);
+		// stdout: `PnPDeviceID : USB\VID_0BRA&PID_8387\00C0CA7B646E`
+		const result = /(?<=PnPDeviceID : ).*/.exec(stdout);
+
+		if (!result) {
+			throw new Error('Couldn\'t find a Wi-Fi device');
+		}
 
-	const result = /Hardware Port: Wi-Fi\nDevice: (?<device>en\d)/.exec(stdout);
-	if (!result) {
-		throw new Error('Couldn\'t find a Wi-Fi device');
+		return result[0];
 	}
 
-	return result.groups.device;
+	throw new Error('macOS or Windows only');
 });
 
 const isOn = async device => {
-	const {stdout} = await execa('ifconfig', [device]);
-	return stdout.includes('status: active');
+	if (process.platform === 'darwin') {
+		const {stdout} = await execa('ifconfig', [device]);
+		return stdout.includes('status: active');
+	}
+
+	if (process.platform === 'win32') {
+		const {stdout} = await powershell('Get-NetAdapterAdvancedProperty', ['-Name Wi-Fi -RegistryKeyword RFOff -AllProperties | Format-List -Property RegistryValue']);
+		// stdout: `RegistryValue : {0}`
+		const status = /(?<={).+?(?=})/.exec(stdout);
+		return !Number.parseInt(status, 10);
+	}
 };
 
 const toggleDevice = async (device, turnOn) => {
-	await execa('networksetup', ['-setairportpower', device, (turnOn ? 'on' : 'off')]);
+	if (process.platform === 'darwin') {
+		await execa('networksetup', [
+			'-setairportpower',
+			device,
+			turnOn ? 'on' : 'off'
+		]);
+	} else if (process.platform === 'win32') {
+		const setStatus = turnOn ? 0 : 1;
+		console.log('Testing...'); // Odd Issue: Without this log statement, tests fail on Windows 10
+		await powershell('Start-Process PowerShell -Verb RunAs -WindowStyle Hidden', [`-ArgumentList "Set-NetAdapterAdvancedProperty -Name Wi-Fi -RegistryKeyword RFOff -AllProperties -RegistryValue ${setStatus}"`]);
```

Comment:
I don't know PowerShell, but is `Start-Process PowerShell -Verb RunAs -WindowStyle Hidden` really a valid command? Everything after `Start-Process` look like arguments.

Reply:
You are right. I've addressed this on the latest commit.